### ISSUE TITLE
`LocalFinalVariableName` must include `_`

### ISSUE THREAD

**Issue by boris-petrov** [checkstyle/checkstyle#15012]
I have read check documentation: https://checkstyle.sourceforge.io/checks/naming/localfinalvariablename.html#LocalFinalVariableName
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="LocalFinalVariableName"/>
  </module>
</module>

/var/tmp $ cat Test.java
public class Test {
	public void foo() {
		try (var _ = lock()) {
		} catch (Exception exception) {
		}
	}

	public AutoCloseable lock() {
		return null;
	}
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/boris/Test.java:3:26: Name '_' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalFinalVariableName]
Audit done.
Checkstyle ends with 1 errors.
```

`_` is a valid identifier in Java 22 and should be supported by Checkstyle. I fixed it locally by using `<property name="format" value="^(_|[a-z][a-zA-Z0-9]*)$"/>`.

**Comment by romani** [comment]
@nrmancuso , can it be done during summer?

**Comment by mahfouz72** [comment]
Thank you for your report your issue and alot more will be done as a part of summer project. Please let us know if you run into something similar realted to java 21/22 features

**Comment by rnveach** [comment]
@mahfouz72 This seems like a sub-task of https://github.com/checkstyle/checkstyle/issues/14942 .

**Comment by mahfouz72** [comment]
@rnveach @nrmancuso please approve
